File: Linguaggio C/Esercizi Funzioni/Anni.c
/**
 * Anni.c
 */
#include <stdio.h>

int day(int giorno, int mese, int anno);

/**
 * Anno bisestile:
 * 	-> Anni non secolari divisibili per 4
		Anni secolari divisibili per 400
		Gli anni secolari sono quelli divisibili per 100
 */

int main()
{
	int giorno, mese, anno;

	printf("Inserire una data nel formato gg/mm/aaaa: ");
	scanf("%d/%d/%d", &giorno, &mese, &anno);

	int g_totali = day(giorno, mese, anno);
	printf("\nLa data fornita indica il giorno numero %d dell'anno!!!\a\a\a", g_totali);

	return 0;
}

int day(int giorno, int mese, int anno)
{
	int x = 0;
	int giorni = 0;

	if(anno%100 == 0 && anno%400 == 0)
	{
		x = 1;
	}
	else if(anno%4 == 0)
	{
		x = 1;
	}
	else
	{
		x = 0;
	}

	if(x == 1)
	{
		for(int i = 1; i<mese; ++i)
		{
			if(i==1 || i==3 || i==5 || i==7 || i==8 || i==10 || i==12)
			{
				giorni = giorni + 31;
			}
			else if(i == 2)
			{
				giorni = giorni + 29;
			}
			else
			{
				giorni = giorni + 30;
			}
		}
		giorni = giorni + giorno;
	}
	else
	{
		for(int i = 1; i<mese; ++i)
		{
			if(i==1 || i==3 || i==5 || i==7 || i==8 || i==10 || i==12)
			{
				giorni = giorni + 31;
			}
			else if(i == 2)
			{
				giorni = giorni + 28;
			}
			else
			{
				giorni = giorni + 30;
			}
		}
		giorni = giorni + giorno;
	}
	return giorni;
}

File: Python/Altro/Anni.py
#Anni.py
#
#Programma per contare il giorno inserito da terminale
#e dirti il numero del giorno dell'anno inserito

def giornoDellAnno(giorno, mese, anno):
	x = 0
	giorni = 0

	if (anno % 100 == 0) and (anno % 400 == 0):
		x = 1
	elif anno%4 == 0:
		x = 1
	else: x = 0

	if x == 1:

		i = 1
		while i < mese:
			if i==1 or i==3 or i==5 or i==7 or i==8 or i==10 or i==12:
				giorni = giorni + 31
			elif i==2:
				if x==1:
					giorni = giorni + 29
				else: giorni = giorni + 28
			else: giorni = giorni + 30
			i = i+1
		giorni = giorni + giorno
	return giorni

giorno = input("Inserire il giorno: ")
mese = input("Inserire il mese: ")
anno = input("Inserire l'anno: ")

g_totali = giornoDellAnno(int(giorno), int(mese), int(anno))
print("Il giorno che hai inserito è il", g_totali, "-esimo dell'anno!!")


